### PR TITLE
 BaSP-MR-20: Get-Create-activity

### PR DESCRIPTION
**Pull request by TomasRomanoo**

### CODE REVIEW

**Review by julianv97 — APPROVED** [review]

File: src/controllers/activity.js [hunk 1]
```diff
@@ -1,5 +1,46 @@
...
 
+const getAllActivity = async (req, res) => {
+  try {
+    const activity = await Activity.find({ isActive: true });
+    res.status(200).json({
+      message: 'Complete Activity list',
+      data: activity,
+      error: false,
+    });
+  } catch (error) {
```
Comment (FrancoLelli) [inline src/controllers/activity.js:11]:
```suggestion
  } catch (error) {
    res.status(500).json({
      message: error,
      data: undefined,
      error: true,
    });
```

**Review by FrancoLelli — CHANGES_REQUESTED** [review]
Provide more information about the errors

File: src/routes/activity.js [hunk 2]
```diff
@@ -5,6 +5,8 @@ import validations from '../validations/activity';
 const activityRouter = Router();
 
 activityRouter
+  .get('/', activityController.getAllActivity)
+  .post('/', validations.validateUpdate, activityController.createActivity)
```
Comment (Tilajc) [inline src/routes/activity.js:9]: do you have a function to validateCreate? the function Is almost the same, but the values should have require
```suggestion
  .post('/', validations.validateCreate, activityController.createActivity)
```

**Review by Tilajc — CHANGES_REQUESTED** [review]
the code is almost approved, but I don't know what validateUpdate does, but you should use a different function for create and update

File: src/routes/activity.js (hunk 2 above)
Comment (FrancoLelli) [reply src/routes/activity.js:9]: Does not exist validateCreate. 
You would have to create the function and which fields are required

Comment (TomasRomanoo) [reply src/routes/activity.js:9]: I delete the validateCreate because it was the same of validateUpdate.

Comment (Tilajc) [reply src/routes/activity.js:9]: validateCreate should have .required in every field, that is the difference

Comment (Tilajc) [reply src/routes/activity.js:9]: when you create an activity, every field should be completed

**Review by Tilajc — APPROVED** [review]
Well Done!

**Pull request merged**